added cgroup updates and oomKillDisable annotation

pdgetrf opened the pull request:
(1) added updating pod level cgroups aggregration based on pod resource spec
(2) added oomKillDisable annotation, passed down to pod level cgroup file memory.oom_control

Tested on a local environment

vinaykul commented on pkg/kubelet/cm/pod_container_manager_linux.go:
```diff
@@ -75,21 +75,35 @@ func (m *podContainerManagerImpl) Exists(pod *v1.Pod) bool {
+	}
+
+	// parse oomKillDisable flag
+	oomKillDisable := pod.ObjectMeta.Annotations["oomKillDisable"]
```
lets define this as a const in some location

pdgetrf replied:
fixed

vinaykul commented on pkg/kubelet/cm/types.go:
```diff
@@ -35,6 +35,8 @@ type ResourceConfig struct {
 	HugePageLimit map[int64]int64
 	// Maximum number of pids
 	PodPidsLimit *int64
+	// OOM Kill Configuration
```
Lets clarify the comment as something like "Configuration of out-of-memory process kill setting for pod"

pdgetrf replied:
fixed

vinaykul reviewed (changes requested):
Couple of minor changes

Commit:
(1) added updating pod level cgroups aggregration based on pod resource spec
(2) added oomKillDisable annotation, passed down to pod level cgroup file memory.oom_control

Tested on a local environment
